### ISSUE TITLE
Compiled minified Angular App - Unknown expression "all". If you wanted a literal array, use ["literal", [...]]

### ISSUE THREAD

**Issue by vittrup88** [openlayers/ol-mapbox-style#280]
When running local with npm start the style is applied to to vector tile layer succesfully.

When compiled and running in production we get this error:
Unknown expression "all". If you wanted a literal array, use ["literal", [...]]
![image](https://user-images.githubusercontent.com/13432911/86584748-b10a0180-bf85-11ea-9490-fd263c35feea.png)

When debugging I find there is something missing in the registry. The compiled version looks like this:
![image](https://user-images.githubusercontent.com/13432911/86584977-fc241480-bf85-11ea-921a-ede364115962.png)

But when running local registry has alot more functions:
![image](https://user-images.githubusercontent.com/13432911/86585054-1cec6a00-bf86-11ea-8907-f0c75bccd5d0.png)

How does this happen? - Do I miss an import or compile options ? It looks like it is an error in mapbox-gl-stylespec

**Comment by ahocevar** [comment]
This error comes from mapbox-gl-stylespec's expression parser. Maybe your style has an invalid expression? Verify that your style is valid, using [gl-style-validate](https://www.npmjs.com/package/@mapbox/mapbox-gl-style-spec#gl-style-validate).

**Comment by vittrup88** [comment]
I have now validated the style with no errors. I just don't get why this is compiled and has missing functions in mapbox-gl-stylescpec expression parser. Do you think I'm missing something in tsconfig.json or any simular?

tsconfig:
`{
  "compileOnSave": true,
  "compilerOptions": {
    "baseUrl": "./",
    "outDir": "./dist/out-tsc",
    "sourceMap": true,
    "declaration": false,
    "moduleResolution": "node",
    "emitDecoratorMetadata": true,
    "experimentalDecorators": true,
    "target": "ES5",
    "allowJs": true,
    "importHelpers": false,
    "paths": {
      "@angular/*": ["../node_modules/@angular/*"]
    },
    "typeRoots": [
      "node_modules/@types"
    ],
    "lib": [
      "es2016",
      "dom"
    ],
    "module": "es2015",
    "plugins": [
      { "transform": "ts-transformer-keys/transformer" }
    ]
  }
}`

The style:
`{
  "version": 8,
  "name": "DataQuality-Pipe-WasteWater",
  "sources": {
    "wastewater_network": {
      "tiles": [
        "dummyurl"
      ],
      "type": "vector"
    }
  },
  "layers": [
    {
      "id": "PipeDimmed-Utility-Other",
      "type": "line",
      "source": "wastewater_network",
      "source-layer": "dataApex_WasteWaterPipe",
      "filter": [
        "all",
        [
          "!=",
          "ownerid",
          1601
        ],
        [
          "==",
          "transportid",
          2001
        ]
      ],
      "layout": {
        "visibility": "visible"
      },
      "paint": {
        "line-color": {
          "property": "ownerid",
          "stops": [
            [
              1602,
              "#00AFAF"
            ],
            [
              1610,
              "#969696"
            ]
          ],
          "type": "categorical"
        },
        "line-translate-anchor": "map",
        "line-width": {
          "property": "categorytypeid",
          "stops": [
            [
              401,
              2.6
            ],
            [
              402,
              2.2
            ],
            [
              403,
              1.8
            ]
          ],
          "type": "categorical"
        },
        "line-dasharray": []
      }
    },
    {
      "id": "PipeDimmed-Utility-Other-dashed",
      "type": "line",
      "source": "wastewater_network",
      "source-layer": "dataApex_WasteWaterPipe",
      "filter": [
        "all",
        [
          "!=",
          "ownerid",
          1601
        ],
        [
          "!=",
          "transportid",
          2001
        ]
      ],
      "layout": {
        "visibility": "visible"
      },
      "paint": {
        "line-color": {
          "property": "ownerid",
          "stops": [
            [
              1602,
              "#00AFAF"
            ],
            [
              1610,
              "#969696"
            ]
          ],
          "type": "categorical"
        },
        "line-translate-anchor": "map",
        "line-width": {
          "property": "categorytypeid",
          "stops": [
            [
              401,
              2.6
            ],
            [
              402,
              2.2
            ],
            [
              403,
              1.8
            ]
          ],
          "type": "categorical"
        },
        "line-dasharray": [
          6,
          2,
          2,
          2
        ]
      }
    },
    {
      "id": "PipeDimmed-Utility-Non-Lateral",
      "type": "line",
      "source": "wastewater_network",
      "source-layer": "dataApex_WasteWaterPipe",
      "filter": [
        "all",
        [
          "==",
          "ownerid",
          1601
        ],
        [
          "==",
          "transportid",
          2001
        ],
        [
          "!=",
          "categorytypeid",
          403
        ]
      ],
      "layout": {
        "visibility": "visible"
      },
      "paint": {
        "line-color": {
          "property": "subnetworktypeid",
          "stops": [
            [
              211,
              "#DE0000"
            ],
            [
              212,
              "#00AF00"
            ],
            [
              213,
              "#0000FF"
            ]
          ],
          "type": "categorical"
        },
        "line-translate-anchor": "map",
        "line-width": {
          "property": "categorytypeid",
          "stops": [
            [
              401,
              2.6
            ],
            [
              402,
              2.2
            ],
            [
              403,
              1.8
            ]
          ],
          "type": "categorical"
        },
        "line-dasharray": []
      }
    },
    {
      "id": "PipeDimmed-Utility-Non-Lateral-dashed",
      "type": "line",
      "source": "wastewater_network",
      "source-layer": "dataApex_WasteWaterPipe",
      "filter": [
        "all",
        [
          "==",
          "ownerid",
          1601
        ],
        [
          "!=",
          "transportid",
          2001
        ],
        [
          "!=",
          "categorytypeid",
          403
        ]
      ],
      "layout": {
        "visibility": "visible"
      },
      "paint": {
        "line-color": {
          "property": "subnetworktypeid",
          "stops": [
            [
              211,
              "#DE0000"
            ],
            [
              212,
              "#00AF00"
            ],
            [
              213,
              "#0000FF"
            ]
          ],
          "type": "categorical"
        },
        "line-translate-anchor": "map",
        "line-width": {
          "property": "categorytypeid",
          "stops": [
            [
              401,
              2.6
            ],
            [
              402,
              2.2
            ],
            [
              403,
              1.8
            ]
          ],
          "type": "categorical"
        },
        "line-dasharray": [
          6,
          2,
          2,
          2
        ]
      }
    },
    {
      "id": "PipeDimmed-Utility-Non-Lateral-403",
      "type": "line",
      "source": "wastewater_network",
      "source-layer": "dataApex_WasteWaterPipe",
      "filter": [
        "all",
        [
          "==",
          "ownerid",
          1601
        ],
        [
          "==",
          "categorytypeid",
          403
        ]
      ],
      "layout": {
        "visibility": "visible"
      },
      "paint": {
        "line-color": {
          "property": "subnetworktypeid",
          "stops": [
            [
              211,
              "#DE0000"
            ],
            [
              212,
              "#00AF00"
            ],
            [
              213,
              "#0000FF"
            ]
          ],
          "type": "categorical"
        },
        "line-translate-anchor": "map",
        "line-width": 1.6,
        "line-dasharray": [
          4,
          2
        ]
      }
    }
  ]
}`

**Comment by ahocevar** [comment]
Maybe you are using an old version of ol-mapbox-style that uses an outdated expression parser? Make sure you're using the latest: ol-mapbox-style@6.1.2.

The error you get is not a JavaScript error, it is the expression parser not understanding your style.

**Comment by vittrup88** [comment]
> Maybe you are using an old version of ol-mapbox-style that uses an outdated expression parser? Make sure you're using the latest: ol-mapbox-style@6.1.2.
> 
> The error you get is not a JavaScript error, it is the expression parser not understanding your style.

I'm using this in package.json
    "ol-mapbox-style": "^6.1.2"

I'm a bit confused since it all works when I'm running the solution locally. Then there is no error. Is there a way to print versions of ol-mapbox-style and mapbox-gl-stylespecs to be sure what versions are used. It just seems something has been skipped maybe in default functions?

**Comment by ahocevar** [comment]
You can do

    npm ls ol-mapbox-style
    npm ls mapbox-gl-stylespec

to get the versions of the respective packages.

**Comment by vittrup88** [comment]
I do get this:
npm ls ol-mapbox-style
`-- @apx10/map@14.3.4
  `-- ol-mapbox-style@6.1.2

npm ls mapbox-gl-style-spec
`-- (empty)

It seems I dont get mapbox-gl-style-spec. This is compiled with ol-mapbox-style ? Do i need a reference ?

**Comment by ahocevar** [comment]
That looks good. No idea if your problem is related to Angular's compilation process.

**Comment by vittrup88** [comment]
I found the problem is in angular compilation. 
In my angular.json I had to set buildOptimizer to false.

I tested with ng build --prod and here I had to set the optimizer to false else I get this error.
As I understand it - the buildoptimizer should compress the build even more and remove unused code. In this situation some functions are removed from the registry of filter function. This is a mapbox style specs issue, but I'm not sure if this is fixable?
